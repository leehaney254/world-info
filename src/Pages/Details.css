.HomeFlag {
  width: 40%;
  height: 30%;
}

.detailContainer {
  display: flex;
  justify-content: space-between;
  padding: 2rem 1rem;
  border-bottom: 1px solid white;
}

#detailsFlag {
  display: flex;
  justify-content: center;
}

#detailscolor {
  color: white;
}
