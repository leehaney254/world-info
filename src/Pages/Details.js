import React from 'react';
import Navbar from '../components/Navbar';

const Details = () => (
  <div>
    <Navbar data="Town/City info" />
    <main>
      <div>
        image
        <div>
          <p>kenya</p>
          <p>Currency</p>
        </div>
      </div>
      <div>
        <div>
          <p>Capital</p>
          <p>Nairobi</p>
        </div>
        <div>
          <p>Area</p>
          <p>580367.0</p>
        </div>
        <div>
          <p>Population</p>
          <p> 53771300</p>
        </div>
        <div>
          <p>timezones</p>
          <p>UTC+03:00</p>
        </div>
      </div>
    </main>
  </div>
);

export default Details;
