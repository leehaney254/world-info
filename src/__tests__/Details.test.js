import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { Provider } from 'react-redux';
import store from '../redux/configStore';
import Details from '../Pages/Details';

configure({ adapter: new Adapter() });

describe('Details component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Provider store={store}><Details /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
