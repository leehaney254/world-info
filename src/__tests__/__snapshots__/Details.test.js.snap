// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Home component renders correctly has a match with snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Home
    countryData={
      Object {
        "country": Array [
          Object {
            "area": 580367,
            "capital": "Nairobi",
            "countryName": "Kenya",
            "currencies": "Kenyan shilling",
            "flag": "https://flagcdn.com/ke.svg",
            "population": 53771300,
            "region": "Africa",
            "timezones": "UTC+03:00",
          },
        ],
        "error": null,
        "loading": false,
      }
    }
  />
</ContextProvider>
`;
